common/src/binary/BinaryBuffer.cpp

```diff
@@ -227,4 +253,28 @@ uint16_t BinaryBuffer::get() {
     }
 }
 
+template <>
+uint8_t BinaryBuffer::get() {
+    if (view_) {
+        std::array<ByteType, sizeof(uint8_t)> tmp;
+        std::transform(buffer_view_.begin(), buffer_view_.end(), tmp.begin(), [](const auto &entry) {
+            return *entry;
+        });
+        return *reinterpret_cast<const uint8_t *>(tmp.data());
+    } else {
+        return *reinterpret_cast<const uint8_t*>(buffer_.data());
+    }
+}
+
+template <>
+std::string BinaryBuffer::get() {
```

Comment:
This implementation presumes that char is one byte of size. Even if this is almost always true, I think that `static_assert` won't hurt.